### DISCUSSION TITLE
Why winforms doesn't work on platforms other than windows?

### DISCUSSION THREAD

**Discussion by Maniues** [dotnet/winforms#4718] [Q&A] [answered]
Why winforms doesn't work on platforms other than windows?

**Comment by RussKie** [comment, marked as answer]
For the most part Windows Forms is a wrapper around Win32 API (which essentially make Windows). 

Mono has made a relatively successful attempt to provide matching implementations for those Win32 API on non-Windows platforms, and it was/is (?) possible to run Windows Forms apps with [Mono Windows Forms]( https://github.com/mono/mono/tree/master/mcs/class/System.Windows.Forms) on non-Windows platforms. It is also possible to run Windows app under Wine, though I can't comment how well they work.
